c2rust-ast-builder/src/builder.rs

```diff
@@ -252,47 +247,6 @@ impl Make<Extern> for Abi {
     }
 }
 
-impl<'a> Make<Mutability> for &'a str {
-    fn make(self, _mk: &Builder) -> Mutability {
-        match self {
-            "" | "imm" | "immut" | "immutable" => Mutability::Immutable,
-            "mut" | "mutable" => Mutability::Mutable,
-            _ => panic!("unrecognized string for Mutability: {:?}", self),
-        }
-    }
-}
-
-impl<'a> Make<Unsafety> for &'a str {
-    fn make(self, _mk: &Builder) -> Unsafety {
-        match self {
-            "" | "safe" | "normal" => Unsafety::Normal,
-            "unsafe" => Unsafety::Unsafe,
-            _ => panic!("unrecognized string for Unsafety: {:?}", self),
-        }
-    }
-}
-
-impl<'a> Make<Constness> for &'a str {
-    fn make(self, _mk: &Builder) -> Constness {
-        match self {
-            "" | "normal" | "not-const" => Constness::NotConst,
-            "const" => Constness::Const,
-            _ => panic!("unrecognized string for Constness: {:?}", self),
-        }
-    }
-}
-
-impl<'a> Make<UnOp> for &'a str {
-    fn make(self, _mk: &Builder) -> UnOp {
-        match self {
-            "deref" | "*" => UnOp::Deref(Default::default()),
-            "not" | "!" => UnOp::Not(Default::default()),
-            "neg" | "-" => UnOp::Neg(Default::default()),
```

Comment:
I'm not sure getting rid of all of these is helpful.  `"!"` is pretty easy to read and understand (although things like `"not"` and the other non-standard spellings for things seem bad).  It has the downside of not being as easily searchable since it's not tied to a symbol like `UnOp::Not`, but it's also easier to read and write in a bunch of ways.